setup/requirements-check.py

```diff
@@ -260,19 +260,27 @@ def get_version(self, package):
         return None
 
 
+def _strip_comment(line):
+    return line.split('#', 1)[0].strip()
+
+
 def parse_requirements(reqpath: Path) -> Dict[str, List[Tuple[str, Marker]]]:
     """ Parses a requirement file to a dict of {package: [(version, markers)]}
 
     The env markers express *whether* that specific dep applies.
     """
     reqs = {}
     with reqpath.open('r', encoding='utf-8') as f:
-        for requirement in pkg_resources.parse_requirements(f):
+        for req_line in f:
+            req_line = _strip_comment(req_line)
+            if not req_line:
+                continue
+            requirement = Requirement(req_line)
             version = None
-            if requirement.specs:
-                if len(requirement.specs) > 1:
-                    raise NotImplementedError('mutli spec not supported yet')
-                version = requirement.specs[0][1]
+            if requirement.specifier:
+                if len(requirement.specifier) > 1:
+                    raise NotImplementedError('multi spec not supported yet')
+                version = list(requirement.specifier)[0].version
```

Comment:
Should this check that `operator` is `==` ?

Reply:
I don't think so, this is script compares versions from multiple sources. The constraint is checked during manual review.